File: december/findJudge.ts
/*
 * @Author: your name
 * @Date: 2021-12-19 22:12:45
 * @LastEditTime: 2021-12-20 09:58:11
 * @LastEditors: Please set LastEditors
 * @Description: 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
 * @FilePath: /myweekend/leetcode刷题笔记/december/findJudge.ts
 */

function findJudge(n : number,trust : number[][]) : number {
    let outDegree : number[] = []; //出度
    let inDegree : number[] = []; //入度
    let res = -1;
    for (let index = 0; index < n; index++) {
        outDegree.push(0);
        inDegree.push(0);
    }
    
    trust.forEach(element => {
        outDegree[element[0]-1] += 1;
        inDegree[element[1]-1] += 1
    });

    for (let index = 0; index < n; index++) {
        if ( outDegree[index] == 0 && inDegree[index] == n-1){
            if(res == -1){
                res = index+1;
            }else {
                return -1;
            }
        }
    }
    return res;

}

(()=>{
    let res = findJudge(2,[[1,2]]);
    console.log(res);
})();
export {};

File: december/findJudge.py
'''
Author: Ambrose
Date: 2021-12-19 19:15:40
LastEditTime: 2021-12-19 21:56:48
LastEditors: Please set LastEditors
Description: 打开koroFileHeader查看配置 
FilePath: /myweekend/leetcode刷题笔记/december/findJudge.py
'''

'''
description: 
param {int} n
param {list} trust
return {*}
'''
def findJudge(n: int, trust) -> int:
    # dic = [[[1,4],[2,3]],[[2],[3]]]
    res = -1
    dic = [[] for i in range(n)]
    for item in dic: 
        item.append([])
        item.append([])
    print(dic)
    for item in trust:
        index = item[0]-1
        aim = item[1]-1
        dic[index][1].append(aim)
        dic[aim][0].append(index)
    print(dic)
    for i in range(len(dic)):
        if(len(dic[i][1]) == 0 and len(dic[i][0]) == n-1):
            if (res == -1):
                res = i+1
            else:
                return -1
        
    return res

if __name__ == '__main__':
    print(findJudge(3, [[1,2],[2,3]]))

    pass
